build/AzurePipelinesTemplates/WindowsAppSDK-StaticValidationTest-Stage.yml

```diff
@@ -0,0 +1,118 @@
+parameters:
+- name: "SignOutput"
+  type: boolean
+  default: false
+  
+stages:
+- stage: StaticValidationTest
+  dependsOn: Pack
+  condition: not(failed())
+  jobs:
+  - job: StaticValidationTest
+    pool:
+      type: windows
+      isCustom: true
+      name: 'ProjectReunionESPool-2022' # This stage is purely for ES, hence this custom pool
+    timeoutInMinutes: 120
+    steps:
+    - task: NuGetToolInstaller@1
+    - task: NuGetAuthenticate@1
+
+    - task: powershell@2
+      displayName: 'DevCheck: Setup/Verify development environment'
+      inputs:
+        targetType: filePath
+        filePath: tools\DevCheck\DevCheck.ps1
+        arguments: -NoInteractive -Offline -Verbose -CheckTestPfx -Clean -CheckDependencies -ShowSystemInfo
+        workingDirectory: '$(Build.SourcesDirectory)'
+
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download Foundation'
+      inputs:
+        source: specific
+        runVersion: specific
+        project: $(System.TeamProjectId)
+        pipeline: $(_useBuildOutputFromPipeline)
+        pipelineId: $(_useBuildOutputFromBuildId)
+        artifactName: "TransportPackage"
+        targetPath: $(System.ArtifactsDirectory)
+
+    - task: CopyFiles@2
+      inputs:
+        SourceFolder: '$(System.ArtifactsDirectory)'
+        Contents: | 
+          **/*.nupkg
+        targetFolder: $(Build.SourcesDirectory)\localpackages
+
+    - task: PowerShell@2
+      name: ExtractWindowsAppSDKVersion
+      displayName: Extract WindowsAppSDKVersion
+      inputs:
+        targetType: 'inline'
+        script: |
+          $files = Get-ChildItem "$(Build.SourcesDirectory)\localpackages" -Filter "*Microsoft.WindowsAppSDK.Foundation.TransportPackage*.nupkg"
+          Write-Host $files.Count
+          if ($files.Count -eq 1)
+          {
+              Write-Host "file:" $files.FullName
+              $version = (($files.FullName -Split "Microsoft.WindowsAppSDK.Foundation.TransportPackage.")[1] -Split ".nupkg")[0]
+              Write-Host "|$version|"
+              Write-Host "##vso[task.setvariable variable=FoundationVersion]$version"
+          }
+          else
+          {
+              Write-Host "Error: Found more than one Foundation TransportPackage Found"
+              Write-Host "Please make sure only one package is present"
+              Write-Host "##vso[task.complete result=Failed;]DONE"
+          }
+
+    - task: VSBuild@1
+      displayName: 'Build PackageInspectionTest.sln'
```

Comment:
Nonblocking suggestion: consider adding "(restore)" to distinguish between these 2 tasks?

Reply:
Good idea